### ISSUE TITLE
AutoMM: Make it obvious how many GPUs being used / if they are detected

### ISSUE THREAD

**Issue by Innixma** [autogluon/autogluon#3284]
Currently it is not sufficiently clear if GPU is being used nor how many of them are being used.
While there is some logging from pytorch lightning, it is not sufficient for novice users, nor to help us debug user issues.

We should be logging information by default such as:

- [ ] number of GPUs available
- [ ] number of GPUs used
- [ ] Name of GPU (if possible): ex, "RTX 4090"
- [ ] GPU VRAM
- [ ] AutoGluon version
- [ ] CUDA version
- [ ] PyTorch version

Mainline logs from [quick start tutorial](https://auto.gluon.ai/dev/tutorials/multimodal/multimodal_prediction/multimodal-quick-start.html): 

```
No path specified. Models will be saved in: "AutogluonModels/ag-20230605_214930/"
AutoGluon infers your prediction problem is: 'binary' (because only two unique label-values observed).
	2 unique label values:  [0, 1]
	If 'binary' is not the correct problem_type, please manually specify the problem_type parameter during predictor init (You may specify problem_type as one of: ['binary', 'multiclass', 'regression'])
INFO:lightning_fabric.utilities.seed:Global seed set to 0
AutoMM starts to create your model. ✨

- Model will be saved to "/home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930".

- Validation metric is "roc_auc".

- To track the learning progress, you can open a terminal and launch Tensorboard:
    ```shell
    # Assume you have installed tensorboard
    tensorboard --logdir /home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930
    ```

Enjoy your coffee, and let AutoMM do the job ☕☕☕ Learn more at https://auto.gluon.ai

INFO:pytorch_lightning.utilities.rank_zero:Using 16bit None Automatic Mixed Precision (AMP)
INFO:pytorch_lightning.utilities.rank_zero:GPU available: True (cuda), used: True
INFO:pytorch_lightning.utilities.rank_zero:TPU available: False, using: 0 TPU cores
INFO:pytorch_lightning.utilities.rank_zero:IPU available: False, using: 0 IPUs
INFO:pytorch_lightning.utilities.rank_zero:HPU available: False, using: 0 HPUs
INFO:pytorch_lightning.accelerators.cuda:LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
INFO:pytorch_lightning.callbacks.model_summary:
  | Name              | Type                | Params
----------------------------------------------------------
0 | model             | MultimodalFusionMLP | 198 M 
1 | validation_metric | BinaryAUROC         | 0     
2 | loss_func         | CrossEntropyLoss    | 0     
----------------------------------------------------------
198 M     Trainable params
0         Non-trainable params
198 M     Total params
396.017   Total estimated model params size (MB)
INFO:pytorch_lightning.utilities.rank_zero:Epoch 0, global step 1: 'val_roc_auc' reached 0.44559 (best 0.44559), saving model to '/home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930/epoch=0-step=1.ckpt' as top 3
INFO:pytorch_lightning.utilities.rank_zero:Epoch 0, global step 4: 'val_roc_auc' reached 0.67913 (best 0.67913), saving model to '/home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930/epoch=0-step=4.ckpt' as top 3
INFO:pytorch_lightning.utilities.rank_zero:Epoch 1, global step 5: 'val_roc_auc' reached 0.68527 (best 0.68527), saving model to '/home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930/epoch=1-step=5.ckpt' as top 3
INFO:pytorch_lightning.utilities.rank_zero:Epoch 1, global step 8: 'val_roc_auc' reached 0.75223 (best 0.75223), saving model to '/home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930/epoch=1-step=8.ckpt' as top 3
INFO:pytorch_lightning.utilities.rank_zero:Epoch 2, global step 9: 'val_roc_auc' reached 0.77288 (best 0.77288), saving model to '/home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930/epoch=2-step=9.ckpt' as top 3
INFO:pytorch_lightning.utilities.rank_zero:Time limit reached. Elapsed time is 0:02:00. Signaling Trainer to stop.
INFO:pytorch_lightning.utilities.rank_zero:Epoch 2, global step 11: 'val_roc_auc' reached 0.76521 (best 0.77288), saving model to '/home/ci/autogluon/docs/tutorials/multimodal/multimodal_prediction/AutogluonModels/ag-20230605_214930/epoch=2-step=11.ckpt' as top 3
Start to fuse 3 checkpoints via the greedy soup algorithm.
AutoMM has created your model 🎉🎉🎉

```

**Comment by zhiqiangdon** [comment]
- We can add more logging info. Where do you recommend to log them?
- `INFO:pytorch_lightning.accelerators.cuda:LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]` tells the number of GPUs used.
- CUDA VRAM is nontrivial to get.

**Comment by Innixma** [comment]
> * We can add more logging info. Where do you recommend to log them?
> * `INFO:pytorch_lightning.accelerators.cuda:LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]` tells the number of GPUs used.
> * Not sure how to compute the CUDA VRAM before using it.

- Where to log: Somewhere before training the model.
- the PyTorch Lightning log is not novice friendly. It requires non-trivial knowledge to understand what it means. For example, it could be easy to misinterpret this log as "0 CUDA devices found"
- CUDA VRAM: Have you tried the following suggestions? https://stackoverflow.com/questions/58216000/get-total-amount-of-free-gpu-memory-and-available-using-pytorch

**Comment by zhiqiangdon** [comment]
Resolved by #3291
